### PR TITLE
feat: Iceberg/create-catalog

### PR DESCRIPTION
**Pull request by ClSlaid**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR implements:
1. the creation of iceberg catalog, from SQL.
2. list database under iceberg catalog. (return database struct)
3. list table under iceberg database. (return table struct)

Note:
- moved catalog option validating from `intepreter` to `binder`
- stashed some unrelated logic of reading iceberg tables.
- `allow_insecure` will also affect the catalog!

## Example

```sql
-- create on aws s3
CREATE CATALOG iceberg_ctl
  TYPE=ICEBERG
  CONNECTION=( 
    URI="s3://my_bucket/path/to/db" 
    AWS_KEY_ID="<access-key>"
    AWS_SECRET_KEY="<secret_key>"
    SESSION_TOKEN="<session_token>"
  );

SHOW DATABASES IN iceberg_ctl;
-- +--------------------------+
-- | databases_in_iceberg_ctl |
-- +--------------------------+
-- | iceberg_db               |
-- +--------------------------+
-- 1 row in set
-- Time: 0.066s

SHOW TABLES IN iceberg_ctl.iceberg_db;
-- +----------------------+
-- | tables_in_iceberg_db |
-- +----------------------+
-- | iceberg_tbl          |
-- +----------------------+
-- 1 row in set
-- Time: 0.144s
```

## Future Possibility

The metadata deserialize is mostly done with a [fork](https://github.com/ClSlaid/iceberg-rs/tree/feat/v1/manifests) of [iceberg-rs](https://docs.rs/iceberg-rs/0.1.0/iceberg_rs/). To fully implement iceberg in Databend repository may result in large size of code and drags compile time.

It may be a better choice to have a standalone `databend-iceberg` crate, working as a query engine, implementing validation, reading, writing, and partitioning for iceberg. And then plug it in Databend.

### CODE REVIEW

File: src/query/service/src/catalogs/catalog_manager.rs [hunk 1]
```diff
@@ -22,11 +22,16 @@ use common_config::CatalogConfig;
 use common_config::Config;
 use common_exception::ErrorCode;
 use common_exception::Result;
-use common_meta_app::schema::CatalogType;
+use common_meta_app::schema::CatalogOption;
 use common_meta_app::schema::CreateCatalogReq;
 use common_meta_app::schema::DropCatalogReq;
+use common_meta_app::schema::IcebergCatalogOption;
+#[cfg(feature = "iceberg")]
```
Comment (Xuanwo) [inline src/query/service/src/catalogs/catalog_manager.rs:29]: `iceberg` doesn't need a new feature.

File: src/query/storages/iceberg/src/catalog.rs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    pub fn try_create(name: &str, flatten: bool, operator: DataOperator) -> Result<Self> {
+        Ok(Self {
+            name: name.to_string(),
+            flatten,
+            operator,
+        })
+    }
+
+    /// list read databases
+    pub async fn list_database_from_read(&self, tenant: &str) -> Result<Vec<Arc<dyn Database>>> {
```
[the remark this answers is not in the document]
Comment (ClSlaid) [reply src/query/storages/iceberg/src/catalog.rs:103]: This is a placeholder to fill `TableMeta` and `DatabaseMeta`.

But noticing `Hive` just fill them up with `Default::default()`, aligning with `Hive` might be better.

File: src/query/storages/iceberg/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[package]
+name = "common-storages-iceberg"
+version = "0.1.0"
```
Comment (BohuTANG) [inline src/query/storages/iceberg/Cargo.toml:3]: We can inherited these cargo infos from workspace.

**Comment by BohuTANG** [comment]
Is it possible add a test for this PR?

**Comment by Xuanwo** [comment]
> Working on it, I'm implementing new SQLs for listing databases and tables under different catalogs.

Please do this in another new PR.

**Comment by ClSlaid** [comment]
> > Working on it, I'm implementing new SQLs for listing databases and tables under different catalogs.
> 
> Please do this in another new PR. The SQL itself also needs tests.

Sure it will be.

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -47,7 +47,3 @@
 /docker/               @datafuselabs/databend-tool-reviewer
 /scripts/              @datafuselabs/databend-tool-reviewer
 /.github/              @datafuselabs/databend-tool-reviewer
-
-# Dependences
```
Comment (Xuanwo) [inline .github/CODEOWNERS]: Strange change. Something wrong?

**Comment by Xuanwo** [comment]
Hello, is this PR still running?

**Comment by ClSlaid** [comment]
> Hello, is this PR still running?

Sorry for leaving it to stale, haven't been running it for a long time. I may reopen it when I'm really ready.

**Comment by ClSlaid** [comment]
图片

This might work.

**Comment by sundy-li** [comment]
@ClSlaid  Forgot to tell you that we support decimal types now!

**Comment by ClSlaid** [comment]
Strange failures in sql logic tests, having been fixing it...

**Review by Xuanwo — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]
👍

File: src/query/storages/iceberg/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+common-exception = { path = "../../../common/exception" }
+common-expression = { path = "../../expression" }
+common-meta-app = { path = "../../../meta/app" }
+common-meta-types = { path = "../../../meta/types" }
+common-storage = { path = "../../../common/storage" }
+
+async-trait = "0.1"
+chrono = { workspace = true }
+futures = "0.3"
+iceberg-rs = { git = "https://github.com/clslaid/iceberg-rs", branch = "feat/v1/manifests" }
```
Comment (BohuTANG) [inline src/query/storages/iceberg/Cargo.toml:22]: Please move the fork crate to https://github.com/datafuse-extras

Comment (Xuanwo) [reply src/query/storages/iceberg/Cargo.toml:22]: @ClSlaid please visit @PsiACE for help.

Comment (ClSlaid) [reply src/query/storages/iceberg/Cargo.toml:22]: Transfer finished, will submit another PR to redirect.

**Pull request merged**